src/OnChain.ts

```diff
@@ -98,23 +98,31 @@ export const OnChainMixin = (superclass) =>
       return fee
     }
 
-    async onChainPay({ address, amount, memo }: IOnChainPayment): Promise<ISuccess> {
+    async onChainPay({ address, amount, memo, sendAll = false }: IOnChainPayment): Promise<ISuccess> {
       let onchainLogger = this.logger.child({
         topic: "payment",
         protocol: "onchain",
         transactionType: "payment",
         address,
         amount,
         memo,
+        sendAll,
       })
 
-      if (amount <= 0) {
-        const error = "Amount can't be negative"
-        throw new ValidationError(error, { logger: onchainLogger })
-      }
+      if (!sendAll) {
+        if (amount <= 0) {
+          const error = "Amount can't be negative"
```

Comment:
"Amount can't be negative, and can only be zero if sendAll = true" may be more appropriate here.

Reply:
But that is the branch if sendAll = false, which I just kept the original flow intact.

Reply:
yup I know it's part of the original flow. but now that we have another branch where `amount = 0` as part of the flow, I think it's worth being a bit more specific on the error (which was already a bit too vague to get started, as the error message mention negative result but the check is also actually negative or zero).

Reply:
Made the update.